### ISSUE TITLE
Dynamic Assembly memory leak

### ISSUE THREAD

**Issue by udlose** [barnhill/barcodelib#142]
There is a memory leak in the implementation of `XmlSerializer` usage.  A new `XmlSerializer(typeof(SaveData))` is created every time the following are called:
- `BarcodeLib.ToXML()`
- `BarcodeLib.FromXML()`

This usage creates a dynamic assembly every time the `XmlSerailizer(typeof(T))` constructor is invoked. This results in a memory leak.

**EDIT:** _This usage creates a dynamic assembly every time the `XmlSerailizer(typeof(T))` constructor is invoked._ This is not correct. The leak occurs in different constructors.  I've been so used to having to correct these in other apps, I confused the dynamic assembly memory leak with a different constructor. My apologies.

**Comment by barnhill** [comment]
Destruction of the serializer still leaks?

**Comment by udlose** [comment]
I clarified the issue and corrected a statement I made.  And in the case of the other (leaky) constructors (which you are not using here), yes, the assembly still leaks.  Dynamically created assemblies aren't cleaned up until the app domain is unloaded even if the object that created them has been GC'd - dynamically created assemblies are not considered a rooted reference.

Have a look here for more info: [Dynamically Generated Assemblies](https://docs.microsoft.com/en-us/dotnet/api/system.xml.serialization.xmlserializer?view=netframework-4.7.2#dynamically-generated-assemblies)

All this being said, creating an `XmlSerializer` instance is a little bit heavier than other framework types.  And since you are always creating an instance that serializes the same type `SaveData`, it makes sense to create it once and hold onto it.
